src/tools/compiletest/src/runtest.rs

```diff
@@ -1627,14 +1624,12 @@ impl<'test> TestCx<'test> {
                     rustc.args(&["--error-format", "json"]);
                 }
             }
-            Ui => {
-                // In case no "--error-format" has been given in the test, we'll compile
-                // a first time to get the compiler's output then compile with
-                // "--error-format json" to check if all expected errors are actually there
-                // and that no new one appeared.
-                if !self.props.disable_ui_testing_normalization {
-                    rustc.arg("-Zui-testing");
-                }
+            Ui => if !self.props
+                 .compile_flags
+                 .iter()
+                 .any(|s| s.starts_with("--error-format"))
+            {
+                rustc.args(&["--error-format", "json"]);
```

Comment:
`-Zui-testing` was lost during revert.

Reply:
It doesn't work at all. Whatever I try to do, I always end with empty stderr so I'll keep my code for the moment.

Reply:
Ah nevermind, I didn't update correctly...